packages/fxa-auth-server/lib/routes/emails.js

```diff
@@ -529,69 +521,11 @@ module.exports = (
                 }
 
                 // Any matching code verifies the account
-                return db
-                  .verifyEmail(account, account.emailCode)
-                  .then(() => {
-                    const geoData = request.app.geo;
-                    const country =
-                      geoData.location && geoData.location.country;
-                    const countryCode =
-                      geoData.location && geoData.location.countryCode;
-                    return P.all([
-                      log.notifyAttachedServices('verified', request, {
-                        country,
-                        countryCode,
-                        email: account.email,
-                        locale: account.locale,
-                        marketingOptIn: marketingOptIn ? true : undefined,
-                        newsletters,
-                        service,
-                        uid,
-                        userAgent: request.headers['user-agent'],
-                      }),
-                      request.emitMetricsEvent('account.verified', {
-                        // The content server omits marketingOptIn in the false case.
-                        // Force it so that we emit the appropriate newsletter state.
-                        marketingOptIn: marketingOptIn || false,
-                        newsletters,
-                        uid,
-                      }),
-                      reminder
-                        ? request.emitMetricsEvent(
-                            `account.reminder.${reminder}`,
-                            { uid }
-                          )
-                        : null,
-                      verificationReminders.delete(uid),
-                    ]);
-                  })
-                  .then(() => {
-                    // send a push notification to all devices that the account changed
-                    request.app.devices.then(devices =>
-                      push.notifyAccountUpdated(uid, devices, 'accountVerify')
-                    );
-                  })
-                  .then(() => {
-                    // Our post-verification email is very specific to sync,
-                    // so only send it if we're sure this is for sync.
-                    if (service === 'sync') {
-                      const mailOptions = {
-                        acceptLanguage: request.app.acceptLanguage,
-                        service,
-                        uid,
-                      };
-
-                      if (style) {
-                        mailOptions.style = style;
-                      }
-
-                      return mailer.sendPostVerifyEmail(
-                        [],
-                        account,
-                        mailOptions
-                      );
-                    }
-                  });
+                return signupUtils.verifyAccount(
```

Comment:
The more I see of these various "utils" functions, the more I think they should be methods on proper domain objects in the `lib/` directory. Like this could be `account.verify`, then there could be an `account.signin` or whatever too...